src/main/java/appeng/container/implementations/ContainerMEMonitorable.java

```diff
@@ -141,6 +146,16 @@ protected ContainerMEMonitorable(final InventoryPlayer ip, final ITerminalHost m
                 this.addSlotToContainer(this.cellView[y]);
             }
         }
+        if (isAPatternTerminal()) {
```

Comment:
Just a comment - I was wondering why you needed to do this when this should really be done in the inheritors' constructor (i.e. `ContainerPatternTermEx` for example, then realized they each inherit directly from `ContainerMEMonitorable`, and my assumption is that this is to reduce some boilerplate. The issue is that now every `ContainerMEMonitorable` will have these methods that should really only belong to the pattern terminals.

For now this is fine. I'll look into seeing if we can change the inheritance tree so it makes more sense in the future.

Reply:
Yes, extracting a common parent for the terminals parts, containers and GUIs certainly makes sense, but it can't be done w/o changing the old classes and when adding a new terminal I didn't want to touch existing one, it was one of the my first changes in AE2.